kvlds/serialize.c

```diff
@@ -440,8 +440,11 @@ serialize_size(struct node * N)
 		size += SERIALIZE_PERCHILD;
 	}
 
+	/* Sanity check: we can't store more than this in the node. */
+	assert(size <= UINT32_MAX);
```

Comment:
Do we need to wrap this in `#if SIZE_MAX > UINT32_MAX` to avoid stupid compiler warnings on 32-bit platforms?

Reply:
I'll confirm, but I'm sure that we will.

Reply:
The `-Wtautological-constant-out-of-range-compare` warning is really finicky.  On a 32-bit system, only the very last line fails!

```
size_t size;
assert(size <= UINT32_MAX);
assert(size <= UINT32_MAX + 10000);
assert(size <= SIZE_MAX);
assert(size <= SIZE_MAX + 100000);
assert(size <= UINT64_MAX*0.999);
assert(size <= UINT64_MAX);
```

With that in mind, I'm not going to bother wrapping the assert in `#if SIZE_MAX > UINT32_MAX`.  If and when compilers get pickier, we can add such things then.